Source/WebCore/css/parser/CSSPropertyParser.cpp

```diff
@@ -6364,6 +6364,53 @@ bool CSSPropertyParser::consumeOffset(bool important)
     return m_range.atEnd();
 }
 
+bool CSSPropertyParser::consumeListStyleShorthand(bool important)
+{
+    auto& valuePool = CSSValuePool::singleton();
+    bool hasParsedImage = false; // `none` may be image & type, distinguish them.
+    RefPtr<CSSValue> listStylePosition;
+    RefPtr<CSSValue> listStyleImage;
+    RefPtr<CSSValue> listStyleType;
+
+    for (size_t i = 0; i < 3 && !m_range.atEnd(); ++i) {
+        if (auto position = parseSingleValue(CSSPropertyListStylePosition, CSSPropertyListStyle)) {
+            if (listStylePosition)
+                return false;
+            listStylePosition = position;
+            continue;
+        }
+        if (auto image = parseSingleValue(CSSPropertyListStyleImage, CSSPropertyListStyle); image && !hasParsedImage) {
+            if (listStyleImage)
+                return false;
+            listStyleImage = image;
+            hasParsedImage = true;
```

Comment:
Isn't hasParsedImage bit unnecessary? You can just test for non-null listStyleImage. Also the if branch never gets hit due to !hasParsedImage test?